lib/ListView.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import './EditListview.dart';
+import './model.dart';
+import './Constants.dart';
+
+class Listview extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Colors.blue[300],
+        title: Center(
+          child: Text('TIG169 TODO', textAlign: TextAlign.center),
+        ),
+
+        // Drop down menu
+        actions: <Widget>[
+          PopupMenuButton<String>(onSelected: (choice) {
+            var state = Provider.of<MyState>(context, listen: false);
+            state.filterChange(choice);
+          }, itemBuilder: (BuildContext context) {
+            return Constants.choices.map((String choice) {
+              return PopupMenuItem<String>(
+                value: (choice),
+                child: Text(choice),
+              );
+            }).toList();
+          })
+        ],
+      ),
+      body: Consumer<MyState>(
+        builder: (context, state, child) => TodoList(state.list),
+      ),
+
+      //FloatingAction Button.
+      floatingActionButton: FloatingActionButton(
+        child: Icon(Icons.add),
+        onPressed: () async {
+          var newTodo = await Navigator.push(
+              context,
+              MaterialPageRoute(
+                  builder: (context) =>
+                      EditListview(TodoTask(message: null, status: false))));
+          if (newTodo != null) {
+            Provider.of<MyState>(context, listen: false).addTodo(newTodo);
+          }
+        },
+      ),
+    );
+  }
+
+  void choiceAction(String choice) {
+    if (choice == Constants.all) {
+      print('all');
+      Filter.show = "all";
+      print(Filter.show);
+    } else if (choice == Constants.done) {
+      print('done');
+      Filter.show = "done";
+      print(Filter.show);
+    } else if (choice == Constants.undone) {
+      print('undone');
+      Filter.show = "undone";
+      print(Filter.show);
+    }
```

Comment:
Så jag gjorde var att beroende på vilket värde som skrevs till filter (choice i ditt fall) så returnerades en lista.

  List<ToDoModel> filteredList(String filter) {
    if (filter == "Done") {
      return _list.where((todo) => todo.isDone == true).toList();
    } else if (filter == "Not Done") {
      return _list.where((todo) => todo.isDone == false).toList();
    }

Din kod skiljer sig dock på andra sätt, bl.a hur du har deklarerat din konstanter för drop down menyn. Jag kan tyvärr för lite för att avgöra vad som saknas i ditt fall, och om det går att få det att fungera.